.github/workflows/main.yml

```diff
@@ -0,0 +1,89 @@
+name: CI
+on: [push]
+
+  # A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout Repo
+      - uses: actions/checkout@v2
+
+      # Set up python
+      - name: Set up Python 3.x
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
+
+      # Install Deps
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install tox
+      # Run Check
+      - name: Run Code Linters
+        run: tox -e check
+
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8, 3.9]
+    steps:
+      # Checkout Repo
+      - uses: actions/checkout@v2
+
+      # Set up python
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      # Install Deps
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install tox tox-gh-actions
+      # Run Tests
+      - name: Run Python Tests
+        run: tox
+
+      - name: Show items
+        run: ls -la
+
```

Comment:
I think I had this in the example I showed you. Possibly this was just for debugging purposes, and probably isn't needed here.

```suggestion
```